lib/timers.js

```diff
@@ -763,6 +792,12 @@ function createImmediate(args, callback) {
 exports.clearImmediate = function(immediate) {
   if (!immediate) return;
 
+  if (isPromise(immediate))
+    immediate = immediate.timer;
+
+  if (typeof immediate._onImmediate !== 'function') {
```

Comment:
This code doesn’t have any typechecks right now anyway, you can just pass in anything and have hope it doesn’t go up in flames.